dot/network/transaction_test.go

```diff
@@ -28,32 +29,36 @@ func TestDecodeTransactionHandshake(t *testing.T) {
 	require.Equal(t, testHandshake, msg)
 }
 
+//go:generate mockgen -destination=mock_transaction_handler_test.go -package $GOPACKAGE . TransactionHandler
 func TestHandleTransactionMessage(t *testing.T) {
 	t.Parallel()
 
-	basePath := utils.NewTestBasePath(t, "nodeA")
-	mockhandler := &MockTransactionHandler{}
-	mockhandler.On("HandleTransactionMessage",
-		mock.AnythingOfType("peer.ID"),
-		mock.AnythingOfType("*network.TransactionMessage")).
+	expectedMsgArg := &TransactionMessage{
+		Extrinsics: []types.Extrinsic{{1, 1}, {2, 2}},
+	}
+
+	ctrl := gomock.NewController(t)
+	transactionHandler := NewMockTransactionHandler(ctrl)
+	transactionHandler.EXPECT().
+		HandleTransactionMessage(peer.ID(""), expectedMsgArg).
 		Return(true, nil)
-	mockhandler.On("TransactionsCount").Return(0)
+
+	transactionHandler.EXPECT().TransactionsCount().Return(0).MaxTimes(1)
```

Comment:
nit you could also batch replace all `.MaxTimes(1)` with nothing

```suggestion
	transactionHandler.EXPECT().TransactionsCount().Return(0)
```